Add better Map.put description to collections lesson

brain-geek opened the pull request:
I guess this fixes #3016 .

doomspork commented on lessons/en/basics/collections.md:
```diff
@@ -196,7 +196,7 @@ iex> %{map | foo: "baz"}
 **Note**: this syntax only works for updating a key that already exists in the map! If the key does not exist, a `KeyError` will be raised.
 
-To create a new key, instead use [`Map.put/3`](https://hexdocs.pm/elixir/Map.html#put/3)
+To create a new key, instead use [`Map.put/3`](https://hexdocs.pm/elixir/Map.html#put/3). They are equivalent, however the [`Map.put/3`](https://hexdocs.pm/elixir/Map.html#put/3) adds the new key even if it did not exist in the original map. As seen in this example:
```
If I'm understanding the issue correctly they're looking for something like this:

```suggestion
To add a new key we can instead use [`Map.put/3`](https://hexdocs.pm/elixir/Map.html#put/3), this adds a new key if one does not exist and updates the record if a value exists at that key already.  We can see these behaviors demonstrated in our example:
```

Then maybe we update the example:

```elixir
iex> map = %{hello: "world"}
%{hello: "world"}
# Try updating our map with a new `:foo` key using the `|` method
iex> %{map | foo: "baz"}
** (KeyError) key :foo not found in: %{hello: "world"}
    (stdlib) :maps.update(:foo, "baz", %{hello: "world"})
    (stdlib) erl_eval.erl:259: anonymous fn/2 in :erl_eval.expr/5
    (stdlib) lists.erl:1263: :lists.foldl/3
# Use `Map.put/3` to add our new key and value
iex> Map.put(map, :foo, "baz")
%{foo: "baz", hello: "world"}
# Use `Map.put/3` to update our key
iex> Map.put(map, :foo, "bar")
%{foo: "bar", hello: "world"}
```

Thoughts @brain-geek @kinson?

brain-geek commented:
After rereading the lesson, I've also added two more changes:
- dropping the deprecated behavior with duplicate keys (in current elixir it's a warning `warning: key :foo will be overridden in map`)
- added `Map.get` example and reshuffled order of it with map definitions
